### PR TITLE
Add eager loading for polymorphic belongs_to associations

### PR DESCRIPTION
**Pull request by leoarnold**

### CODE REVIEW

File: lib/config/locales/en.yml [hunk 1]
```diff
@@ -60,14 +60,6 @@ en:
           resolution: "Search for attributes that are in the database or set
             the Mongoid.raise_not_found_error configuration option to false,
             which will cause a nil to be returned instead of raising this error."
-        eager_load:
```
Comment (p-mongo) [inline lib/config/locales/en.yml]: Everything having to do with this exception must remain in Mongoid for backwards compatibility reasons, even if Mongoid no longer uses this exception.

**Comment by leoarnold** [comment]
@p-mongo :heavy_check_mark: done

**Comment by p-mongo** [comment]
Thank you, I have this PR in my todo list.

**Comment by p-mongo** [comment]
This will be merged as https://github.com/mongodb/mongoid/pull/4727.

**Pull request closed without merging**